pai-experimental/cdp.sol

```diff
@@ -333,9 +339,26 @@ contract CDP is DSMath, DSNote, Template {
 
     function terminateBusiness(uint price) public note {
         require(!settlement && price != 0);
+        updateRates();
         settlement = true;
         liquidationPenalty = RAY;
-        collateralSettlementPrice = price;
+        closeTime = block.timestamp;
+        priceOracle.terminate(ASSET_BTC, price);
+        liquidator.settlePhaseOne();
+    }
+
+    function quickLiquidate(uint _num) public note {
+        require(settlement);
+        require(liquidatedCDP != CDPIndex);
+        require(block.timestamp > closeTime + bufferPeriod );
+        uint upperLimit = min(liquidatedCDP + _num, CDPIndex);
+        for(uint i = liquidatedCDP + 1; i <= upperLimit; i++) {
+            if(CDPRecords[i].collateral > 0)
+                liquidateInternal(i);
+        }
+        liquidatedCDP = upperLimit;
+        if(liquidatedCDP == CDPIndex)
+            liquidator.settlePhaseTwo();
     }
```

Comment:
all calculations should use DSMath or SafeMath.

Reply:
need your help